101-webapp-linux-sonarqube-mysql/azuredeploy.json

```diff
@@ -0,0 +1,218 @@
+      }
+    },
+    "databaseDTU": {
+      "type": "int",
+      "allowedValues": [
+        50,
+        100
+      ],
+      "defaultValue": 50,
+      "metadata": {
+        "description": "Azure database for MySQL pricing tier : 50 DTU , 100 DTU"
+      }
+    },
+    "databaseSkuName": {
+      "type": "string",
+      "allowedValues": [
+        "MYSQLB100",
+        "MYSQLB50"
+      ],
+      "defaultValue": "MYSQLB50",
+      "metadata": {
+        "description": "Azure database for MySQL sku name : MYSQLB100 (Basic 100 DTU tier) or MYSQLB50 (Basic 50 DTU tier)"
+      }
+    },
+    "databaseSkuSizeMB": {
+      "type": "int",
+      "allowedValues": [
+        102400,
+        51200
+      ],
+      "defaultValue": 51200,
+      "metadata": {
+        "description": "Azure database for MySQL Sku Size "
+      }
+    },
+    "databaseSkuTier": {
+      "type": "string",
+      "allowedValues": [
+        "Basic"
+      ],
+      "defaultValue": "Basic",
+      "metadata": {
+        "description": "Azure database for MySQL pricing tier"
+      }
+    },
+    "mysqlVersion": {
+      "type": "string",
+      "allowedValues": [
+        "5.6",
+        "5.7"
+      ],
+      "defaultValue": "5.7",
+      "metadata": {
+        "description": "MySQL version"
+      }
+    }
+  },
+  "variables": {
+    "databaseName": "[concat(parameters('siteName'), 'database')]",
+    "serverName": "[concat(parameters('siteName'), 'pgserver')]",
+    "jdbcSonarUserName": "[concat(parameters('administratorLogin'),'@',variables('serverName'))]",
+    "hostingPlanName": "[concat(parameters('siteName'), 'serviceplan')]"
+
+  },
+  "resources": [
+    {
+      "comments":"This is the Linux web app with the Sonarquebe Docker image and the DB",
+      "type": "Microsoft.Web/sites",
+      "name": "[parameters('siteName')]",
+      "properties": {
+        "siteConfig": {
+          "linuxFxVersion": "DOCKER|SONARQUBE"
+        },
+        "name": "[parameters('siteName')]",
+        "serverFarmId": "[variables('hostingPlanName')]",
+        "hostingEnvironment": ""
+      },
+      "apiVersion": "2016-03-01",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[variables('hostingPlanName')]",
+        "[variables('databaseName')]"
+      ],
+      "resources": [
+        {
+          "name": "appsettings",
+          "type": "config",
+          "apiVersion": "2015-08-01",
+          "comments": "This are the generated settings for the Sonar jdbc connectionstring",
+          "dependsOn": [
+            "[resourceId('Microsoft.Web/sites', parameters('siteName'))]"
+          ],
+          "tags": {
+            "displayName": "SonarappSettings"
+          },
+          "properties": {
+            "SONARQUBE_JDBC_URL": "[concat('jdbc:mysql://', variables('serverName') ,'.mysql.database.azure.com:3306/',variables('databaseName'),'?verifyServerCertificate=true&useSSL=true&requireSSL=false&useUnicode=true&characterEncoding=utf8')]",
```

Comment:
can we get this url through the reference function?  hardcoding will prevent the sample for working in other clouds

Reply:
Unfortunately I haven't found any property to get this via a reference function, I suppose it will come in later previews, but I will continue looking for it.